src/basic_recipes/band.jl

```diff
@@ -19,6 +19,16 @@ end
 convert_arguments(P::Type{<: Band}, x::AbstractVector{<:Number}, y::AbstractVector{<:Interval}) =
     convert_arguments(P, x, leftendpoint.(y), rightendpoint.(y))
 
+convert_arguments(P::Type{<:Band},
+                  x::AbstractVector{<:Dates.TimeType},
+                  ylower::AbstractVector{<:Real},
+                  yupper::AbstractVector{<:Real}) =
+    convert_arguments(P, Dates.value.(x), ylower, yupper)
+convert_arguments(P::Type{<:Band},
+                  x::AbstractVector{<:Dates.TimeType},
+                  y::AbstractVector{<:Interval}) =
+    convert_arguments(P, Dates.value.(x), leftendpoint.(y), rightendpoint.(y))
```

Comment:
This effectively just ignores Dates when plotting band. E.g. the added test plots:
![image](https://github.com/user-attachments/assets/b1403e7c-efaf-4a04-a666-872a6995c4ba)

We also don't want to handle Dates, Unitful, etc in convert_arguments. That's what dim_converts are for.